test/pubspec_test.dart

```diff
@@ -654,5 +658,75 @@ web:
         });
       });
     });
+
+    group("features", () {
+      test("can be null", () {
+        var pubspec = new Pubspec.parse('features:', sources);
+        expect(pubspec.features, isEmpty);
+      });
+
+      test("throws if it's not a map", () {
+        expectPubspecException('features: 12', (pubspec) => pubspec.features);
+      });
+
+      test("throws if it has non-string keys", () {
+        expectPubspecException(
+            'features: {1: {}}', (pubspec) => pubspec.features);
+      });
+
+      test("throws if a key isn't a Dart identifier", () {
+        expectPubspecException(
+            'features: {foo-bar: {}}', (pubspec) => pubspec.features);
+      });
+
+      test("allows null values", () {
+        var pubspec = new Pubspec.parse(
+            '''
+features:
+  foobar:
+''',
+            sources);
+        expect(pubspec.features, equals({'foobar': []}));
+      });
+
+      test("throws if the value isn't a map", () {
+        expectPubspecException(
+            'features: {foobar: 1}', (pubspec) => pubspec.features);
+      });
+
+      test("throws if the value's dependencies aren't valid", () {
+        expectPubspecException(
+            '''
+features:
+  foobar:
+    dependencies:
+      baz: not a version range
+''',
+            (pubspec) => pubspec.features);
+      });
+
+      test("parses valid dependency specifications", () {
+        var pubspec = new Pubspec.parse(
+            '''
+features:
+  foobar:
+    dependencies:
+      baz: 1.0.0
```

Comment:
Ah ok - that is probably fine (and I don't think it would be a breaking change to switch it later to support versions everywhere if we decided that was important).